bofire/strategies/predictives/multi_fidelity.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import pandas as pd
+
+from bofire.data_models.features.api import TaskInput
+from bofire.data_models.strategies.predictives.multi_fidelity import (
+    MultiFidelityStrategy as DataModel,
+)
+from bofire.strategies.predictives.sobo import SoboStrategy
+from bofire.utils.naming_conventions import get_column_names
+
+
+class MultiFidelityStrategy(SoboStrategy):
+    def __init__(self, data_model: DataModel, **kwargs):
+        super().__init__(data_model=data_model, **kwargs)
+        self.task_feature_key = self.domain.inputs.get_keys(TaskInput)[0]
+
+        ft = data_model.fidelity_thresholds
+        M = len(self.domain.inputs.get_by_key(self.task_feature_key).fidelities)  # type: ignore
+        self.fidelity_thresholds = ft if isinstance(ft, list) else [ft] * M
+
+    def _ask(self, candidate_count: int) -> pd.DataFrame:
+        """Generate new candidates (x, m).
+
+        This is a greedy optimization of the acquisition function. We first
+        optimize the acqf for the target fidelity to generate a candidate x,
+        then select the lowest fidelity that has a variance exceeding a
+        threshold.
+
+        Args:
+            candidate_count (int): number of candidates to be generated
+
+        Returns:
+            pd.DataFrame: DataFrame with candidates (proposed experiments)
+        """
+        if candidate_count > 1:
+            raise NotImplementedError("Batch optimization is not yet implemented")
+        task_feature: TaskInput = self.domain.inputs.get_by_key(self.task_feature_key)  # type: ignore
+        task_feature.allowed = [True] + [False] * (len(task_feature.categories) - 1)
```

Comment:
I think this line still assumes the fidelities are ordered?

Reply:
Yes, good spot!